sql/src/main/java/io/crate/executor/transport/NodeOperationTreeGenerator.java

```diff
@@ -194,8 +210,17 @@ public Void visitCollect(Collect plan, NodeOperationTreeContext context) {
 
     @Override
     public Void visitMerge(Merge merge, NodeOperationTreeContext context) {
+        Plan subPlan = merge.subPlan();
+
+        boolean useDirectResponse = context.noPreviousPhases() &&
```

Comment:
`noPreviousPhases` needs to be called before the `addPhase` call for this to work.